src/query/api/v1/handler/prometheus/native/middleware.go

```diff
@@ -23,84 +23,52 @@ package native
 
 import (
 	"net/http"
-	"strings"
 	"time"
 
-	"github.com/gorilla/mux"
-	"github.com/jonboulle/clockwork"
 	"go.uber.org/zap"
 
-	"github.com/m3db/m3/src/query/util/logging"
-	"github.com/m3db/m3/src/x/headers"
-	xhttp "github.com/m3db/m3/src/x/http"
-	"github.com/m3db/m3/src/x/instrument"
+	"github.com/m3db/m3/src/query/api/v1/middleware"
 )
 
-// QueryResponseOptions are the options to construct the QueryResponse middleware func.
-type QueryResponseOptions struct {
-	Threshold      time.Duration
-	InstrumentOpts instrument.Options
-	Clock          clockwork.Clock
+// WithQueryParams adds the query request parameters to the middleware options.
+var WithQueryParams middleware.OverrideOptions = func(opts middleware.Options) middleware.Options {
+	opts.Logging.Fields = func(r *http.Request, start time.Time) []zap.Field {
```

Comment:
Is it always ok to blow away whatever logging options may already exist? Maybe we need to do something like capture the existing `opts.Logging.Fields()`, running that method, and  then appending the new fields to those?

Reply:
that's a good point.